soc/nxp/imx/imx9/imx95/m7/soc.c

```diff
@@ -175,6 +177,10 @@ void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 
 	/* Clear PRIMASK */
 	__enable_irq();
+
+#ifdef CONFIG_PM_CPU_SHELL
+	NVIC_EnableIRQ(LPUART3_IRQn);
+#endif
```

Comment:
I don't like it very much. This special handling here looks some sort of architectural violation. This function shouldn't need to know about it, specially because it can get in here without a shell action.